v-next/hardhat-network-helpers/src/internal/network-helpers/network-helpers.ts

```diff
@@ -0,0 +1,281 @@
+   * await networkHelpers.setNextBlockBaseFeePerGas(1000000); // Set base fee to 1,000,000
+   */
+  public setNextBlockBaseFeePerGas(baseFeePerGas: NumberLike): Promise<void> {
+    return setNextBlockBaseFeePerGas(this.#provider, baseFeePerGas);
+  }
+
+  /**
+   * Modifies an account's nonce by overwriting it.
+   *
+   * @param address The address whose nonce is to be changed.
+   * @param nonce The new nonce.
+   * @returns A promise that resolves once the nonce is set.
+   *
+   * @example
+   * const { networkHelpers } = await hre.network.connect();
+   * await networkHelpers.setNonce("0x123...", 10); // Set the nonce of the account to 10
+   */
+  public setNonce(address: string, nonce: NumberLike): Promise<void> {
+    return setNonce(this.#provider, address, nonce);
+  }
+
+  /**
+   * Sets the PREVRANDAO value of the next block.
+   *
+   * @param prevRandao The new PREVRANDAO value to use.
+   * @returns A promise that resolves once the PREVRANDAO value is set.
+   *
+   * @example
+   * const { networkHelpers } = await hre.network.connect();
+   * await networkHelpers.setPrevRandao(123456789); // Set the PREVRANDAO value
+   */
+  public setPrevRandao(prevRandao: NumberLike): Promise<void> {
+    return setPrevRandao(this.#provider, prevRandao);
+  }
+
+  /**
+   * Writes a single position of an account's storage.
+   *
+   * @param address The address where the code should be stored.
+   * @param index The index in storage.
+   * @param value The value to store.
+   * @returns A promise that resolves once the storage value is set.
+   *
+   * @example
+   * const { networkHelpers } = await hre.network.connect();
+   * await networkHelpers.setStorageAt("0x123...", 0, 0x0000...);
+   */
+  public setStorageAt(
+    address: string,
+    index: NumberLike,
+    value: NumberLike,
+  ): Promise<void> {
+    return setStorageAt(this.#provider, address, index, value);
+  }
+
+  /**
+   * Stops Hardhat Network from impersonating the given address.
+   *
+   * @param address The address to stop impersonating.
+   * @returns A promise that resolves once the impersonation is stopped.
+   *
+   * @example
+   * const { networkHelpers } = await hre.network.connect();
+   * await networkHelpers.stopImpersonatingAccount("0x123...");
+   */
+  public stopImpersonatingAccount(address: string): Promise<void> {
+    return stopImpersonatingAccount(this.#provider, address);
+  }
+
+  /**
+   * Takes a snapshot of the blockchain state at the current block.
+   * @returns A promise that resolves to a `SnapshotRestorer` object, which contains a `restore` method to reset the network to this snapshot.
+   *
+   * @example
+   * const { networkHelpers } = await hre.network.connect();
+   * const snapshot = await networkHelpers.takeSnapshot();
+   * await snapshot.restore(); // Restores the blockchain state
+   */
+  public takeSnapshot(): Promise<SnapshotRestorer> {
+    return takeSnapshot(this.#provider);
+  }
+}
```

Comment:
Let's add `loadFixture` like this here too, instead of exporting it on a separate file. The reason being that now we'll have multiple providers.